autoextract_poet/inputs.py

```diff
@@ -0,0 +1,20 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+
+@dataclass
+class ResponseData:
```

Comment:
Naming question. Two options:

1. Kind explicit in the class name. `ResponseData` (for scrapy data), `AutoExtractResponseData`, `SplashResponseData`
2. Use the module name to spot the kind `scrapy.ResponseData`, `autoextract.ResponseData`, `splash.ResponseData`

You choose 2. I think I agree, but not totally sure. Let's see if the rest have any opinion.

Reply:
I see an appeal of `autoextract.ResponseData`, but on the other hand, it has bitten me in past. Consider how it can be used. This is nice:

```py
from web_poet.inputs import autoextract

class Foo:
    data: autoextract.ResponseData
```

But there is also "autoextract" package (via https://github.com/scrapinghub/scrapinghub-autoextract), so it gets ugly when you need to import both:

```py
import autoextract
from web_poet.inputs import autoextract  # what to do?

# option 1:
from web_poet.inputs.autoextract import ResponseData  # clash with other ResponseData types

# option 2:
import web_poet.inputs.autoextract
data = web_poet.inputs.autoextract.ResponseData  # super verbose
```

You can also have `import autoextract` in `web_poet.inputs.__init__.py`; this allows the following:

```py
from web_poet import inputs
data = inputs.autoextract.ResponseData
```

which is less verbose, and not bad actually - though people will be importing ResponseData directly, making the code harder to read.

Alternative is, as @ivanprado said, to use `AutoExtractResponseData` name instead. I don't have a strong opinion on `AutoExtractResponseData` vs `inputs.autoextract.ResponseData`, but other options look worse to me (`ResponseData`, `web_poet.inputs.autoextract.ResponseData` and `autoextract.ResponseData`).

Reply:
Renamed to `AutoExtractResponseData` as it doesn't hurt :)

Reply:
Actually, we're only using this `ResponseData` class within our own code. As we're planning to move this to `web_poet`, I've changed this back to `ResponseData` and we should use it as a namespace as suggested by @ivanprado and @kmike.